lib/game_presenter.rb

```diff
@@ -0,0 +1,70 @@
+class GamePresenter
+    attr_accessor :presenter
+
+    def initialize(presenter:) 
+        @presenter = presenter
+    end
+
+    def get_input(input=nil)
+        presenter.get_input(input)
+    end
+
+    def output_message(message)
+        presenter.output_message(message)
+    end
+
+    def clear()
+        presenter.clear()
+    end
+end
+
+class ConsoleIO
+    def get_input(input=nil)
+        return gets.chomp
+    end
+
+    def output_message(message)
+        puts message
+    end
+
+    def clear()
+        print "\e[2J\e[f"
+    end
+end
+
+class TestIO
```

Comment:
Why are we nesting three classes in here? Is this just used for testing purposes?

Reply:
I've been keeping duck type classes with the interface it works with in one file. I wasn't sure how its usually organized, so kept it together for easy changes

Reply:
And I just moved TestIO into the spec folder, since its pretty much used only on the testing side